src/sentry/incidents/logic.py

```diff
@@ -642,11 +647,11 @@ def update_alert_rule(
     if include_all_projects is not None:
         updated_fields["include_all_projects"] = include_all_projects
 
-    with transaction.atomic():
-        incidents = Incident.objects.filter(alert_rule=alert_rule).exists()
-        if incidents:
-            snapshot_alert_rule(alert_rule)
+    incidents = Incident.objects.filter(alert_rule=alert_rule).exists()
+    if incidents:
+        snapshot_alert_rule(alert_rule)
```

Comment:
I guess the risk of moving this out of the transaction is that all the incidents might end up closed, but the rule doesn't get updated. Maybe it is actually safer if we have a few seconds delay on the task and keep this in the transaction?

Reply:
I actually forget why I moved it out of the transaction. I think it was an issue when the task was still in the transaction...but I tried that and the tests still pass. All tests are passing with it in the transaction. Can you think of something I'm missing?

Reply:
Oh, just thinking maybe it was an issue before I started using with self.tasks() in the test.